File: app/src/main/java/nl/rsdt/japp/jotial/maps/wrapper/google/GooglePolyline.java
package nl.rsdt.japp.jotial.maps.wrapper.google;

import com.google.android.gms.maps.model.LatLng;

import java.util.List;

import nl.rsdt.japp.jotial.maps.wrapper.IPolyline;

/**
 * Created by mattijn on 08/08/17.
 */

public class GooglePolyline implements IPolyline {
    private final com.google.android.gms.maps.model.Polyline googlePolyline;

    public GooglePolyline(com.google.android.gms.maps.model.Polyline polyline) {
        googlePolyline  = polyline;
    }

    @Override
    public void remove() {
        googlePolyline.remove();
    }

    @Override
    public List<LatLng> getPoints() {
        return googlePolyline.getPoints();
    }

    @Override
    public void setPoints(List<LatLng> points) {
        googlePolyline.setPoints(points);
    }

    @Override
    public void setVisible(boolean visible) {
        googlePolyline.setVisible(visible);
    }
}

File: app/src/main/java/nl/rsdt/japp/jotial/maps/wrapper/google/GooglePolyline.kt
package nl.rsdt.japp.jotial.maps.wrapper.google

import com.google.android.gms.maps.model.LatLng

import nl.rsdt.japp.jotial.maps.wrapper.IPolyline

/**
 * Created by mattijn on 08/08/17.
 */

class GooglePolyline(private val googlePolyline: com.google.android.gms.maps.model.Polyline) : IPolyline {

    override var points: MutableList<LatLng>
        get() = googlePolyline.points
        set(points) {
            googlePolyline.points = points
        }

    override fun remove() {
        googlePolyline.remove()
    }

    override fun setVisible(visible: Boolean) {
        googlePolyline.isVisible = visible
    }
}

